waspc/src/Wasp/Analyzer/Parser/AST.hs

```diff
@@ -7,23 +7,28 @@ module Wasp.Analyzer.Parser.AST
   )
 where
 
+import Wasp.Analyzer.Parser.Ctx
 import Wasp.AppSpec.ExtImport (ExtImportName (..))
 
-newtype AST = AST {astStmts :: [Stmt]} deriving (Eq, Show)
+newtype AST = AST {astStmts :: [WithCtx Stmt]} deriving (Eq, Show)
 
--- Decl <declType> <name> <body>
-data Stmt = Decl Identifier Identifier Expr deriving (Eq, Show)
+data Stmt
+  = -- | Decl <declType> <declName> <declBody>
+    Decl Identifier Identifier (WithCtx Expr)
+  deriving (Eq, Show)
 
+{- ORMOLU_DISABLE -}
 data Expr
-  = Dict [(Identifier, Expr)]
-  | List [Expr]
-  | StringLiteral String
+  = Dict           [(Identifier, WithCtx Expr)]
```

Comment:
At the beginning I just added `Ctx` field to each of the data constructors here, but at some point I decided to make things more composable with `WithCtx`.

Only a bit confusing thing is that often in the code I will use name `expr` for both `WithCtx Expr` and for just `Expr`, but I found that to work ok actually, better than constantly trying to keep up with the names and doing smth like `exprWithCtx`.